<a href="https://colab.research.google.com/github/Gabriel-Roledo-ds/Analise-State-of-Data-2025/blob/main/Mercado_de_dados_SoD.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Análise Exploratória e Descritiva — State of Data Brasil 2025/2026

## Estrutura do notebook (CRISP-DM)

1. Business Understanding — objetivos da análise
2. Data Understanding — ingestão, inspeção estrutural, entendimento do padrão de nulos
3. Data Preparation — tratamento e limpeza
4. Evaluation — análise exploratória (univariada e cruzada)
5. Deployment — insights, conclusões e síntese para apresentação

# Business Understanding



## Contexto

Este notebook analisa os dados da pesquisa "State of Data Brasil 2025/2026" (Data
Hackers/Kaggle), que mapeia o perfil profissional, tecnológico e salarial de quem
trabalha ou busca trabalhar com dados no Brasil.

## Objetivo

Realizar uma análise exploratória e descritiva com foco especial em padrões
relevantes para profissionais que estão entrando na área de dados.

## Perguntas que orientam esta análise

1. Quem trabalha com dados no Brasil hoje — idade, formação, escolaridade?
2. Entre quem foi contratado recentemente (dentro do período coberto pela pesquisa),
qual é o perfil de idade, formação e escolaridade? Esse perfil difere do mercado
como um todo?
3. É preciso vir de TI/Computação pra entrar na área?
4. Qual stack (linguagem, banco de dados, cloud, BI) realmente domina o mercado?
5. Como idade, escolaridade e salário evoluem por senioridade?
6. Quais são as características das pessoas que conseguem transicionar de carreira?
7. (extra) O bloco de perguntas 8 tem respostas interessantes sobre o dia a dia de um
cientista de dados — pode ser interessante tirar insights complementares desses dados.

#Data Understanding

## 1. Ingestão e primeira leitura dos dados

In [1]:
from google.colab import drive

drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import pandas as pd

base_state_of_data = "/content/drive/MyDrive/Dados_Estudo/raw/Final Dataset - State of Data 2025-2026 - Kaggle.csv"

df = pd.read_csv(base_state_of_data)

In [3]:
df.shape

(3495, 388)

In [4]:
df.head(10)

,0.a_token,0.d_data/hora_envio,1.a_idade,1.a.1_faixa_idade,1.b_genero,1.c_cor/raca/etnia,1.d_pcd,1.e_experiencia_profissional_prejudicada,1.e.1_Não acredito que minha experiência profissional seja afetada,"1.e.2_Sim, devido a minha Cor/Raça/Etnia",...,8.d.4_Desenvolvendo modelos de Machine Learning com o objetivo de colocar em produção em sistemas (produtos de dados).,"8.d.5_Colocando modelos em produção, criando os pipelines de dados, APIs de consumo e monitoramento.","8.d.6_Cuidando da manutenção de modelos de Machine Learning já em produção, atuando no monitoramento, ajustes e refatoração quando necessário.","8.d.7_Realizando construções de dashboards em ferramentas de BI como PowerBI, Tableau, Looker, Qlik, etc.","8.d.8_Utilizando ferramentas avançadas de estatística como SAS, SPSS, Stata etc, para realizar análises.","8.d.9_Criando e dando manutenção em ETLs, DAGs e automações de pipelines de dados.",8.d.10_Criando e gerenciando soluções de Feature Store e cultura de MLOps.,"8.d.11_Criando e mantendo a infra que meus modelos e soluções rodam (clusters, servidores, API, containers, etc.)",8.d.12_Treinando e aplicando LLM's para solucionar problemas de negócio.,8.d_maior_tempo_gasto_como_ds
0,apeczcwvds75ba5hml4apec2pb35hflw,20/10/2025 10:20:11,35,35-39,Masculino,Branca,Não,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,a27c74vkj7n8vmib88ca27c74sepwcpa,20/10/2025 10:27:09,20,17-21,Masculino,Branca,Não,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2,7xpkv4qj7mzvsxemhpjy7xpkv4qu61oi,20/10/2025 10:28:59,35,35-39,Masculino,Branca,Não,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,ugzyokw0p1or7axwzugzyz4ombwfgzoo,20/10/2025 10:30:24,29,25-29,Masculino,Parda,Não,Não acredito que minha experiência profissiona...,1.0,0.0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,n7aiatzlzonity0sn7aiat75y36iuqni,20/10/2025 10:33:11,18,17-21,Masculino,Preta,Não,Não acredito que minha experiência profissiona...,1.0,0.0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
5,0z4zyz2c8h97geuk0z4in3nhorg4kmkc,20/10/2025 10:35:35,30,30-34,Feminino,Branca,Não,"Sim, acredito que a minha a experiência profis...",0.0,0.0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
6,u5uja0ocgjm6yvncxs8n7u5ujal5nt6i,20/10/2025 10:37:13,28,25-29,Feminino,Amarela,Não,"Sim, acredito que a minha a experiência profis...",0.0,0.0,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,Entrando em contato com os times de negócio pa...
7,p6mjb4z0mqxm2ugxb9xp6mjb4yctum0m,20/10/2025 10:40:39,31,30-34,Masculino,Preta,Prefiro não informar,"Sim, acredito que a minha a experiência profis...",0.0,1.0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
8,f4g04rpgxfp7dnsef4g04rlq5x2e2rth,20/10/2025 10:41:40,39,35-39,Masculino,Parda,Não,"Sim, acredito que a minha a experiência profis...",0.0,1.0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
9,jmzodyrjiaohxyjeujrjmzodyklmbxjv,20/10/2025 10:42:32,29,25-29,Masculino,Branca,Não,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [5]:
df.tail(10)

,0.a_token,0.d_data/hora_envio,1.a_idade,1.a.1_faixa_idade,1.b_genero,1.c_cor/raca/etnia,1.d_pcd,1.e_experiencia_profissional_prejudicada,1.e.1_Não acredito que minha experiência profissional seja afetada,"1.e.2_Sim, devido a minha Cor/Raça/Etnia",...,8.d.4_Desenvolvendo modelos de Machine Learning com o objetivo de colocar em produção em sistemas (produtos de dados).,"8.d.5_Colocando modelos em produção, criando os pipelines de dados, APIs de consumo e monitoramento.","8.d.6_Cuidando da manutenção de modelos de Machine Learning já em produção, atuando no monitoramento, ajustes e refatoração quando necessário.","8.d.7_Realizando construções de dashboards em ferramentas de BI como PowerBI, Tableau, Looker, Qlik, etc.","8.d.8_Utilizando ferramentas avançadas de estatística como SAS, SPSS, Stata etc, para realizar análises.","8.d.9_Criando e dando manutenção em ETLs, DAGs e automações de pipelines de dados.",8.d.10_Criando e gerenciando soluções de Feature Store e cultura de MLOps.,"8.d.11_Criando e mantendo a infra que meus modelos e soluções rodam (clusters, servidores, API, containers, etc.)",8.d.12_Treinando e aplicando LLM's para solucionar problemas de negócio.,8.d_maior_tempo_gasto_como_ds
3485,via88pdugfgccvf3qrvia8d82xiyxqzw,20/10/2025 12:21:38,32,30-34,Masculino,Branca,Não,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3486,s88lf7ln374555zs88lf7y55lvfhjv8p,20/10/2025 12:19:13,33,30-34,Masculino,Branca,Não,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3487,7mk8kjpssef7mk89llakcol0mz76yni2,20/10/2025 12:01:59,20,17-21,Masculino,Branca,Não,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3488,kbb91ctp2625x1qb20kbb91ssah9c3ll,20/10/2025 12:01:05,61,55+,Feminino,Branca,Não,"Sim, acredito que a minha a experiência profis...",0.0,0.0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3489,qd4hcrp570dnwwf9d1ussqd4hclknd5w,20/10/2025 11:32:06,48,45-49,Masculino,Branca,Não,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3490,i0j62lwhuwodn4mppi0jxpd62jwc2u2r,20/10/2025 11:15:21,21,17-21,Masculino,Parda,Não,Não acredito que minha experiência profissiona...,1.0,0.0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3491,hc0ek239suz0o5368o6hc0ek23q30ymn,20/10/2025 11:10:43,22,22-24,Masculino,Branca,Não,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3492,y2bva0w5olmxvo4wy2bva6n4gt9uxhyy,20/10/2025 11:10:26,31,30-34,Masculino,Branca,Não,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3493,lxsjvi0uhm1qlxqnxqglxsjvim5rydhv,20/10/2025 11:10:07,37,35-39,Masculino,Parda,Não,Não acredito que minha experiência profissiona...,1.0,0.0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3494,omjm897gpg57dhomjmmgr25w3gq49pmm,20/10/2025 10:58:56,41,40-44,Masculino,Branca,Não,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [6]:
# Lista todas as colunas do dataset, uma por linha, numeradas
# Objetivo: mapear visualmente os grupos de colunas (perguntas "pai" vs
# alternativas "filhas" de múltipla escolha) antes de qualquer tratamento

for i, col in enumerate(df.columns):
    print(i, col)

0 0.a_token
1 0.d_data/hora_envio
2 1.a_idade
3 1.a.1_faixa_idade
4 1.b_genero
5 1.c_cor/raca/etnia
6 1.d_pcd
7 1.e_experiencia_profissional_prejudicada
8 1.e.1_Não acredito que minha experiência profissional seja afetada
9 1.e.2_Sim, devido a minha Cor/Raça/Etnia
10 1.e.3_Sim, devido a minha identidade de gênero
11 1.e.4_Sim, devido ao fato de ser PCD
12 1.f_aspectos_prejudicados
13 1.f.1_Quantidade de oportunidades de emprego/vagas recebidas
14 1.f.2_Senioridade das vagas recebidas em relação à sua experiência
15 1.f.3_Aprovação em processos seletivos/entrevistas
16 1.f.4_Oportunidades de progressão de carreira
17 1.f.5_Velocidade de progressão de carreira
18 1.f.6_Nível de cobrança no trabalho/Stress no trabalho
19 1.f.7_Atenção dada pelas pessoas diante das minhas opiniões e ideias
20 1.f.8_Relação com outras pessoas da empresa, em momentos de trabalho
21 1.f.9_Relação com outras pessoas da empresa, em momentos de integração e outros momentos fora do trabalho
22 1.g_vive_no_brasil


## Insights primeira visualização



**Volume:** o dataset tem 3.495 respondentes e 388 colunas. Esse número de
colunas é indício de que a pesquisa usa perguntas de múltipla escolha, e cada
alternativa foi expandida em uma coluna própria (dummy). Cada bloco numerado (1 a 8) representa uma seção temática do questionário (perfil, empresa/vaga, gestão, tecnologia, etc.), e dentro de cada seção, as perguntas de múltipla escolha se abrem em colunas filhas.

**Observação 1 — volume de colunas:** olhando os nomes das colunas (ex:
`1.e.1_...`, `1.e.2_...`, `1.e.3_...`), percebe-se um padrão de numeração
hierárquica (pergunta.subpergunta.alternativa). Isso confirma a hipótese:
uma única pergunta de múltipla escolha (ex: "1.e") se desdobra em várias
colunas binárias, uma para cada alternativa possível.

**Observação 2 — padrão de NaN:** em várias linhas (ex: respondentes 0, 1, 2
e 9), todas as colunas a partir de um certo ponto do questionário aparecem
como `NaN` simultaneamente, em bloco — não de forma espalhada/aleatória.
Isso sugere que o `NaN` aqui não representa "não sei" ou dado ausente por
erro de coleta, e sim que o respondente **não viu aquela pergunta**, porque
a pesquisa tem lógica de pulo de seção (skip pattern) — por exemplo, alguém
que respondeu não trabalhar com dados pode não ter sido exposto às perguntas
técnicas mais adiante.

**Implicação para a análise:** se essa hipótese se confirmar, o tratamento de
nulos não pode ser genérico (ex: `dropna()` ou preencher com 0 direto). Será
necessário distinguir "não respondeu porque pulou a seção" de "não respondeu
por outro motivo", já que isso afeta diretamente o denominador correto para
qualquer cálculo de percentual mais adiante na análise.

**Próximo passo:** validar essas duas hipóteses com uma inspeção estrutural
mais formal (dtypes, proporção de nulos por coluna, e checagem por grupos de
colunas relacionadas).

In [7]:
# Tipos de dados presentes no dataset
print(df.dtypes.value_counts())

float64    315
object      71
int64        2
Name: count, dtype: int64


In [8]:
# Proporção de nulos por coluna — as 20 colunas com mais NaN
print(df.isna().mean().sort_values(ascending=False).head(20))

1.h_pais_onde_mora                                                                                                                                                          0.965093
5.b_oportunidade_buscada                                                                                                                                                    0.947639
5.d_experiencia_em_processos_seletivos                                                                                                                                      0.946495
5.c_tempo_em_busca_de_oportunidade                                                                                                                                          0.945064
5.a_objetivo_na_area_de_dados                                                                                                                                               0.881545
6.d_tecnologia_data_lake                                                                       

### Tipos de dados e padrão de nulos

O dataset é composto por 315 colunas `float64`, 71 `object` e 2 `int64`.
A predominância de float64 confirma que a maior parte das colunas (quase 80%)
são variáveis binárias (0/1/NaN) — resultado da decomposição de perguntas de
múltipla escolha em colunas dummy.

O ranking de nulos por coluna reforça essa leitura: os maiores percentuais de
`NaN` aparecem concentrados em seções específicas do questionário (ex: bloco 8
— rotina de Cientista de Dados, ~86%; bloco 5 — busca de oportunidade,
88-95%), e não distribuídos aleatoriamente pelo dataset. Também aparece um
caso de nulo condicional dentro de um bloco geral (`1.h_pais_onde_mora`,
96,5%), sugerindo dependência de uma resposta anterior.

Isso indica que o `NaN`, neste dataset, é majoritariamente estrutural
(skip pattern da pesquisa) — não dado ausente por erro de coleta. Essa
hipótese ainda precisa ser validada formalmente antes de definir a
estratégia de tratamento de nulos.

In [9]:
# Testa se o NaN é consistente dentro de um grupo (ex: bloco 8.b)
grupo_8b = [c for c in df.columns if c.startswith('8.b.')]

# Conta, por linha, quantas colunas do grupo estão nulas
nulos_por_linha = df[grupo_8b].isna().sum(axis=1)

print(nulos_por_linha.value_counts())

14    3005
0      490
Name: count, dtype: int64


In [10]:
# Repete o teste de consistência no bloco 4.c (linguagens de programação)
grupo_4c = [c for c in df.columns if c.startswith('4.c.')]

nulos_por_linha_4c = df[grupo_4c].isna().sum(axis=1)

print(len(grupo_4c))
print(nulos_por_linha_4c.value_counts())

10
0     2096
10    1399
Name: count, dtype: int64


### Validação do padrão de nulos por skip pattern

Repetindo o teste de consistência em dois blocos distintos — 8.b (técnicas
de Data Science, n=490 respostas) e 4.c (linguagens de programação, n=2.096
respostas) — o resultado se repete: dentro de um mesmo grupo de colunas
filhas, cada respondente ou preencheu **todas** as colunas do grupo, ou
**nenhuma**. Não há casos intermediários.

Isso confirma que o `NaN`, nas colunas dummy de múltipla escolha, representa
"não foi exposto à pergunta" (skip pattern da pesquisa) — e não recusa ou
erro de preenchimento em uma alternativa específica.

**Implicação prática para o restante da análise:** ao calcular qualquer
percentual sobre colunas dummy (ex: "% que usa Python"), o denominador correto
não é o total de 3.495 respondentes, e sim o número de pessoas que efetivamente
respondeu aquela seção.